### ISSUE TITLE
Display scaling/position is intermittently off when starting the game in fullscreen

### ISSUE THREAD

**Issue by db47h** [endless-sky/endless-sky#810]
As shown in the screenshot below,when this occurs the game display is slightly smaller than the full screen area and slightly moved to the lower left. Note that the hitbox for the UI elements is not moved accordingly.

![endless_sky_screen_off](https://cloud.githubusercontent.com/assets/879556/13205716/7254a56c-d8ee-11e5-80a5-c9a65960f552.png)

This is on Ubuntu 15.10 with Nvidia drivers 352.63, 6 core CPU, desktop display 1920x1080@60Hz, and default Compiz settings. Happens on game startup about once every five runs, with version 0.8.11 downloaded from Steam and with latest master (a6bed20). Toggling full screen fixes the problem (press F11 twice) until next time.

It seems that it is caused by some interaction between SDL, Unity (or Compiz) and GL: in the screenshot, the black bar at the top is exactly 24 pixels high and the right one is 57 pixels wide, which exactly corresponds to the dimensions of the Unity menu and launcher on my desktop.

So Unity is not playing nice and, when the window is created with SDL_WINDOW_RESIZABLE set (which we obviously cannot remove), the gl viewport gets messed up . Well, I've checked every other coordinate system involved, and the only thing that goes bonkers when this happens is the gl viewport. 

This feels like a race condition and although the fix below works for me, it may only mitigate the issue on different hardware:

```
diff --git a/source/main.cpp b/source/main.cpp
index 34fac90..bbec052 100644
--- a/source/main.cpp
+++ b/source/main.cpp
@@ -201,6 +201,12 @@ int main(int argc, char *argv[])
                
                GameData::LoadShaders();
                
+               // Fix intermittent issue with fullscreen on linux
+               {
+                       int width, height;
+                       SDL_GL_GetDrawableSize(window, &width, &height);
+                       glViewport(0, 0, width, height);
+               }
                
                UI gamePanels;
                UI menuPanels;
```

I'm wondering if other linux users are experiencing this as well and if the above fix works for them.

**Comment by endless-sky** [comment]
I've seen a similar problem on Ubuntu 14.04. For me, what happens is that occasionally when launching in full screen mode, the image looks a tiny bit squished (i.e. not exactly 100% zoom), and as with your bug, mouse clicks on buttons may not work. I haven't been able to track down a root cause, so I usually just fix it by toggling full screen.

**Comment by db47h** [comment]
Yep, that's the same bug, the image is not offset but squished a bit and off-center. Evidence is strongly in favor of Unity forcibly resizing the window (to fit the launcher and top panel) while GL init is happening. This is obviously a race condition with some external thing not playing nice, but resetting the viewport just before the main loop fixes the problem for me. Haven't seen it re-occur with that patch.

**Comment by endless-sky** [comment]
Well, you're change can't do any harm, so I added it in commit 1124419. Hopefully that will fix the issue, or at least make it less frequent.

**Comment by endless-sky** [comment]
Closing this because the fix is included in 0.9.0, which is now released.
